apps/root/src/common/utils/currency.ts

```diff
@@ -40,11 +41,21 @@ const toSignificant = (
 };
 
 export const toSignificantFromBigDecimal = (
-  currency: string,
+  currency: string | undefined,
   significantDigits = 6,
+  threshold = 0.0001,
   format: object = { groupSeparator: '' }
 ): string => {
+  if (isUndefined(currency)) {
+    return '-';
+  }
+
   const quotient = new Decimal(currency).toSignificantDigits(significantDigits);
+
+  if (quotient.lessThan(threshold)) {
```

Comment:
this should be already handled by formatCurrencyAmount

Reply:
The `formatCurrencyAmount` function needs a BigNumber. But price and balanceUsd are decimal numbers, so changing them to BigNumber just to fix their decimals looks hard. Maybe we can change formatCurrencyAmount to take strings too? I solve this in this PR by using a different function for strings